PWGHF/TableProducer/candidateSelectorOmegac0ToOmegaPi.cxx

```diff
@@ -160,409 +182,494 @@ struct HfCandidateSelectorToOmegaPi {
+              registry.fill(HIST("hSelCompetingCasc"), 0);
+            } else {
+              registry.fill(HIST("hSelCompetingCasc"), 1);
+              registry.fill(HIST("hInvMassXiMinus_rej_cut"), candidate.cascRejectInvmass());
+            }
+          }
+
+          // v0&Casc&Omegac ldl selection
+          if ((candidate.v0ldl() < v0ldlmin) || (candidate.cascldl() < cascldlmin) || (candidate.omegacldl() > omegacldlmax)) {
+            resultSelections = false;
+            registry.fill(HIST("hSelV0_Casc_Omegacldl"), 0);
+          } else {
+            registry.fill(HIST("hSelV0_Casc_Omegacldl"), 1);
+          }
+
+          // Omegac ctau selsection
+          if (candidate.ctauOmegac() > ctauOmegacmax) {
+            resultSelections = false;
+            registry.fill(HIST("hSelctauOmegac"), 0);
+          } else {
+            registry.fill(HIST("hSelctauOmegac"), 1);
+          }
+
+          // Chi2Geo/NDF V0&Casc&Omegac selection
+          if ((candidate.v0Chi2OverNdf() > v0Chi2OverNdfmax) || (candidate.cascChi2OverNdf() > cascChi2OverNdfmax) || (candidate.omegacChi2OverNdf() > omegacChi2OverNdfmax)) {
+            resultSelections = false;
+            registry.fill(HIST("hSelChi2GeooverNDFV0_Casc_Omegac"), 0);
+          } else {
+            registry.fill(HIST("hSelChi2GeooverNDFV0_Casc_Omegac"), 1);
+          }
+
+          // Chi2Topo/NDF (chi2TopoV0ToCasc chi2TopoOmegacToPv chi2TopoCascToOmegac chi2TopoCascToPv) selection  (???????????/NDF of which particle????????)
+          if ((candidate.chi2TopoV0ToCasc() > chi2TopoV0ToCascmax) || (candidate.chi2TopoOmegacToPv() > chi2TopoOmegacToPvmax) || (candidate.chi2TopoCascToOmegac() > chi2TopoCascToOmegacmax) || (candidate.chi2TopoCascToPv() > chi2TopoCascToPvmax)) {
+            resultSelections = false;
+            registry.fill(HIST("hSelChi2TopooverNDFV0_Casc_Omegac"), 0);
+          } else {
+            registry.fill(HIST("hSelChi2TopooverNDFV0_Casc_Omegac"), 1);
+          }
+
+          // DecaylengthXY of Omegac&Casc&V0 selection
+          if ((std::abs(candidate.decayLenXYOmegac()) > decayLenXYOmegacmax) || (std::abs(candidate.decayLenXYCasc()) < decayLenXYCascmin) || (std::abs(candidate.decayLenXYLambda()) < decayLenXYLambdamin)) {
+            resultSelections = false;
+            registry.fill(HIST("hSeldecayLenXYOmegac_Casc_V0"), 0);
+          } else {
+            registry.fill(HIST("hSeldecayLenXYOmegac_Casc_V0"), 1);
+          }
+
+          // KFPA cut cosPaCascToOmegac cosPaV0ToCasc
+          if ((candidate.cosPaCascToOmegac() < cosPaCascToOmegacmin) || (candidate.cosPaV0ToCasc() < cosPaV0ToCascmin)) {
+            resultSelections = false;
+            registry.fill(HIST("hSelcosPaCascToOmegac_V0ToCasc"), 0);
+          } else {
+            registry.fill(HIST("hSelcosPaCascToOmegac_V0ToCasc"), 1);
+          }
+
+          if (resultSelections && inputKF) {
+            registry.fill(HIST("hSelKFstatus"), 1);
+          }
+        }
 
```

Comment:
You should check here wheter you are consuming the `HfOmegaKf` table or not. The rest of the code should be agnostic with respect to it, i.e. only here you are adding something specific for the KF particle